added conditional percentage score calculation wfor the percentage bar

kimiweng233 opened the pull request.

Commit:
Calculate adjusted scores from settings

imericxu commented on pages/index.tsx:
```diff
@@ -48,69 +52,90 @@ export default function Home() {
+    if (scores) {
+      let temp = [];
+      for (let i = 0; i < scores.spans[ScoreCategory.toxic].length; i++) {
+        var toxicity;
```
Don’t use `var`; use `let` or `const` instead.

imericxu commented on pages/index.tsx:
```diff
@@ -48,69 +52,90 @@ export default function Home() {
     const scores = data as PerspectiveScores;
     setScores(scores);
     formatForSentencesAnalysis(scores);
```
Don’t call `formatForSentenceAnalysis` in `updateScore`, should be called in a `useEffect`.

imericxu commented on pages/index.tsx:
```diff
@@ -28,6 +28,10 @@ export default function Home() {
 
+  useEffect(() => {
+    formatForSentencesAnalysis(scores);
+  }, [scoreCategoriesSettings, summaryScoreMode]);
```
Missing `scores` dependency

imericxu commented on pages/index.tsx:
Store the calculated scores in a state:

> All the other parts need to use the quick settings to calculate scores. We should keep the original scores as a state, and create another state that automatically recalculates our custom scores every time the settings or scores updates (use the `useEffect` hook).

Also see changes made in #41, especially:

- New model: `lib/models/SummaryScores.ts`
- States: `textFromLastUpdate` and `adjustedScores`

imericxu replied:
There’s also some duplicate code in #41 I kind of copied from your `formatForSentencesAnalysis` function to automatically calculate the percentage.

imericxu commented on pages/index.tsx:
```diff
@@ -48,69 +52,90 @@ export default function Home() {
+    return out;
+  }
+
   const getPercentage = () => {
```
The `getPercentage` function is no longer being used, b/c it’s being automatically calculated in (#41).

imericxu committed:
Change some deps to dev deps

imericxu committed:
Combine all score structures into one file

imericxu committed:
Configure ESLint for TypeScript

imericxu committed:
Fix ESLint errors (strong warnings)

imericxu committed:
Clean up score calculation

- Merge feature into previous pull
- Move calculations into separate file to clean up index.tsx
- #TODO: Refactor formatForSentenceAnalysis

imericxu committed:
Revert onInput to onChange

Also a simple variable call change, I forgot you could use underscores

imericxu committed:
Move const var out of function

imericxu committed:
Fix interval being reset on every keystroke

Using a ref prevents the interval from being recreated.

imericxu committed:
Prevent disabling all categories

imericxu committed:
Replace ** w/ * above useEffects

They're not functions

imericxu committed:
Update documentation

imericxu committed:
Remove unused css file

imericxu committed:
Rename prev to acc for reduce functions

imericxu committed:
Finish score calculations

imericxu committed:
Remove unused variable
